src/views/SignDid/SignDid.tsx

```diff
@@ -1,106 +1,183 @@
 import { useCallback, useRef } from 'react';
 import { browser } from 'webextension-polyfill-ts';
 
+import {
+  RequestForAttestation,
+  Attestation,
+  Credential,
+} from '@kiltprotocol/core';
+
 import * as styles from './SignDid.module.css';
 
 import { Identity } from '../../utilities/identities/types';
-import { usePopupData } from '../../utilities/popups/usePopupData';
 import { getIdentityCryptoFromSeed } from '../../utilities/identities/identities';
-import { isFullDid } from '../../utilities/did/did';
 
-import { useCopyButton } from '../../components/useCopyButton/useCopyButton';
-import { IdentitiesCarousel } from '../../components/IdentitiesCarousel/IdentitiesCarousel';
 import {
   PasswordField,
   usePasswordField,
 } from '../../components/PasswordField/PasswordField';
 import { backgroundSignDidChannel } from '../../channels/SignDidChannels/backgroundSignDidChannel';
 import { SignDidOriginInput } from '../../channels/SignDidChannels/types';
+import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
+import { useCopyButton } from '../../components/useCopyButton/useCopyButton';
+import { LinkBack } from '../../components/LinkBack/LinkBack';
+import { SharedCredential } from '../../utilities/credentials/credentials';
+import { useBooleanState } from '../../utilities/useBooleanState/useBooleanState';
 
 interface Props {
   identity: Identity;
+  popupData: SignDidOriginInput | undefined;
+  onCancel: () => void;
+  credentials?: SharedCredential[];
 }
 
-export function SignDid({ identity }: Props): JSX.Element | null {
+export function SignDid({
+  identity,
+  popupData,
+  onCancel,
+  credentials,
+}: Props): JSX.Element | null {
   const t = browser.i18n.getMessage;
 
-  const error = !isFullDid(identity.did);
-
-  const { origin, plaintext } = usePopupData<SignDidOriginInput>();
-
   const plaintextRef = useRef<HTMLTextAreaElement>(null);
   const copy = useCopyButton(plaintextRef);
 
   const passwordField = usePasswordField();
 
+  const error = useBooleanState();
+
   const handleSubmit = useCallback(
     async (event) => {
       event.preventDefault();
-
-      const { seed } = await passwordField.get(event);
-      const { sign } = await getIdentityCryptoFromSeed(seed);
-
-      await backgroundSignDidChannel.return(sign(plaintext));
-
-      window.close();
+      error.off();
+
+      try {
+        if (!popupData) {
+          throw new Error('No popup data');
+        }
+
+        const { seed } = await passwordField.get(event);
+        const { sign, keystore, didDetails } = await getIdentityCryptoFromSeed(
+          seed,
+        );
+
+        const signature = sign(popupData.plaintext);
+
+        if (!credentials) {
+          await backgroundSignDidChannel.return(signature);
```

Comment:
Would an early return make things clearer here?
```suggestion
          await backgroundSignDidChannel.return(signature);
          window.close();
          return;
```